tools/js2c.py

```diff
@@ -67,6 +67,88 @@ def ReadLines(filename):
   return result
 
 
+def StripComments(source):
+  chars = source.decode('utf-8')
+  result = []
+  index = 0
+  string_mode = None  # None or one of ('\'', '"', '`')
+  escape_mode = False
+
+  while index < len(chars):
+    char = chars[index]
+
+    # TODO(aqrln): support comments inside placeholders
+    # of ES6 template literals
+    if char in ('\'', '"', '`'):
+      if char == string_mode and not escape_mode:
+        string_mode = None
+      elif not string_mode:
```

Comment:
I would prefer explicit check, like `elif string_mode is not None:`. Thats more pythonic and readable as well.